src/data/chats/mod.rs

```diff
@@ -99,13 +100,15 @@ impl Chats {
         self.saved_chats.iter().find(|c| c.borrow().id == chat_id)
     }
 
-    pub fn set_current_chat(&mut self, chat_id: ChatID) {
+    pub fn set_current_chat(&mut self, chat_id: Option<ChatID>) {
```

Comment:
As long as chat switching happens thru this function, the problem should not happen.